py3status/modules/google_calendar.py

```diff
@@ -0,0 +1,507 @@
+
+    def _gstr_to_date(self, date_str):
+        """ Returns a dateime object from a google calendar date string."""
+        return parser.parse(date_str).replace(tzinfo=tzlocal())
+
+    def _gstr_to_datetime(self, date_time_str):
+        """ Returns a datetime object from a google calendar date/time string."""
+        return parser.parse(date_time_str)
+
+    def _datetime_to_str(self, date_time, dt_format):
+        """ Returns a strftime formatted string from a datetime object."""
+        return date_time.strftime(dt_format)
+
+    def _delta_time(self, date_time):
+        """
+        Returns in a dict the number of days/hours/minutes and total minutes
+        until date_time.
+        """
+        now = datetime.datetime.now(tzlocal())
+        diff = date_time - now
+
+        days = int(diff.days)
+        hours = int(diff.seconds / 3600)
+        minutes = int((diff.seconds / 60) - (hours * 60)) + 1
+        total_minutes = int((diff.seconds / 60) + (days * 24 * 60)) + 1
+
+        return {'days': days,
+                'hours': hours,
+                'minutes': minutes,
+                'total_minutes': total_minutes}
+
+    def _format_timedelta(self, index, time_delta, is_current):
+        """
+        Formats the dict time_to containg days/hours/minutes until an event starts
+        into a composite according to time_to_formatted.
+
+        Returns: A formatted composite.
+        """
+        time_delta_formatted = ''
+
+        if time_delta['total_minutes'] <= self.time_to_max:
+            time_delta_formatted = self.py3.safe_format(
+                self.format_timer,
+                {'days': time_delta['days'],
+                 'hours': time_delta['hours'],
+                 'minutes': time_delta['minutes'],
+                 'is_current': is_current})
+
+        return time_delta_formatted
+
+    def _build_response(self):
+        """
+        Builds the composite reponse to be output by the module by looping through all
+        events and formatting the necessary strings.
+
+        Returns: A composite containing the individual response for each event.
+        """
+        responses = []
+        self.event_urls = []
+
+        for index, event in enumerate(self.events):
+            self.py3.threshold_get_color(index + 1, 'event')
+            self.py3.threshold_get_color(index + 1, 'time')
+
+            event_dict = {}
+
+            event_dict['summary'] = event.get('summary')
+
+            if event_dict['summary'] is not None:
+                if event_dict['summary'].lower() \
+                        in map(lambda e: e.lower(), self.blacklist_events):
+                    continue
+
+            event_dict['location'] = event.get('location')
+            event_dict['description'] = event.get('description')
+            self.event_urls.append(event['htmlLink'])
+
+            if event['start'].get('date') is not None:
+                if self.ignore_all_day_events:
+                    continue
```

Comment:
Weak `ignore_all_day_events` checking. We can haz >1 days events.

Reply:
Do you mean we should still show events >1 days even if `ignore_all_day_events` is true?

Reply:
I'll leave that up to @ultrabug. I'm under impression that `all-day event` is all-day event for the given day and does not come with time... and another thing, I think maybe this could be a control placeholder instead of a config, eg `[\?if=!is_allday {event}]`. I'll leave that up to @ultrabug too as he is the one requesting for this feature.

Reply:
Yes, the API returns a `date` string for all-day events, and a `dateTime` string for regular events, which is how I distinguish. An event that takes all day but has a start/end time isn't really an all day event as far as the calendar is concerned.

Reply:
>An all-day event is an activity that lasts 24 hours or longer. Examples of such an event could include a conference, seminar, or vacation. Usually an event occurs once and can last for one or several days. Events do not occupy blocks of time in your calendar. Instead, they appear at the top as banners.

I think this is okay now.